File: stripe/src/main/java/com/stripe/android/exception/RateLimitException.java
package com.stripe.android.exception;

import android.support.annotation.Nullable;

import com.stripe.android.StripeError;

/**
 * An {@link Exception} indicating that too many requests have hit the API too quickly.
 */
public final class RateLimitException extends InvalidRequestException {

    public RateLimitException(
            @Nullable String message,
            @Nullable String param,
            @Nullable String requestId,
            @Nullable StripeError stripeError) {
        super(message, param, requestId, 429, null, null, stripeError, null);
    }
}

File: stripe/src/main/java/com/stripe/android/exception/RateLimitException.kt
package com.stripe.android.exception

import com.stripe.android.StripeError

/**
 * An [Exception] indicating that too many requests have hit the API too quickly.
 */
class RateLimitException(
    stripeError: StripeError? = null,
    requestId: String? = null,
    message: String? = stripeError?.message,
    cause: Throwable? = null
) : StripeException(
    stripeError,
    requestId,
    429,
    cause,
    message
)
